### PR TITLE
Added wireless timestamper and port code

### PR DESCRIPTION
**Pull request by christopher-s-hall**
buildPTPMessage friend declarations and function definition now use
        CommonPort object
processMessage method in PTP message classes now use CommonPort argument
redundant documentation for buildPTPMessage and processMessage removed
        leaving only that in the base class
Add cast to addEventTimerLocked() call (CommonPort::processEvent)
	preventing Windows compiler warnings
Moved all message interval maintenance to CommonPort, also simiplifying
	Signal message processing
Divided FollowUp processing code into two functions for code received on
	an EtherPort or a WirelessPort (as a Vendor Extension)
Modified Windows daemon code to accept a wireless flags
Added Windows wireless adapter abstract class to implement a PROSet-like
	interface for Windows
Implemented an Intel specific Windows wireless adapter class tested on 8260
	hardware with driver version 19.50.1.6

### CODE REVIEW

**Comment by andrew-elder** [comment]
These changes look to be ok to me. Does anyone else have any comments?

**Comment by andrew-elder** [comment]
@christopher-s-hall - I assume you have confirmed that the "wired" version works ok?

**Comment by andrew-elder** [comment]
@christopher-s-hall - are there any special instructions required now that there are options to run wired vs wireless versions on gPTP? And, have those instructions been written up somewhere?

**Comment by christopher-s-hall** [comment]
@andrew-elder I have confirmed that the wired version still works OK. On Linux, I've tested against the an Extreme switch and with 2 I210s. On Windows, I've tested built-in Ethernet vs Extreme switch. On Linux, the command line interface is unchanged because only Windows wireless is supported at this time. 

On Windows, the wireless command line interface is unchanged from the "wireless" branch, but using a single executable for wired and wireless. An additional command line flag to select wireless ("-w") is added.

I would like to add some documentation - where is the best place to do this?

**Comment by andrew-elder** [comment]
@christopher-s-hall , I think doc update would be best in daemons/gptp/README.rst - do you agree?

Thanks for the testing report.

**Comment by andrew-elder** [comment]
Does anyone else have any comments on this pull request?

**Comment by bdthomsen** [comment]
I glanced at it earlier, and didn't note any concerns.  I'll look at it in more detail later today.

File: daemons/gptp/common/ptp_message.cpp [hunk 1]
```diff
@@ -123,8 +124,18 @@ PTPMessageCommon *buildPTPMessage
 	if (!(messageType >> 3)) {
 		int iter = 5;
 		long req = 4000;	// = 1 ms
+
+		eport = dynamic_cast <EtherPort *> ( port );
+		if (eport == NULL)
+		{
+			GPTP_LOG_ERROR
+				( "Recieved Event Message, but port type "
```
Comment (bdthomsen) [inline daemons/gptp/common/ptp_message.cpp]: "Received" is misspelled.

**Pull request merged**